src/tools/rest-backend/conversion_application.cpp

```diff
@@ -0,0 +1,198 @@
+#include <iostream>
+
+#include "conversion_application.hpp"
+#include "model_configformat.hpp"
+#include "model_entry.hpp"
+#include "model_importedconfig.hpp"
+#include "root_application.hpp"
+#include "service.hpp"
+
+namespace kdbrest
+{
+
+/**
+ * @brief the constructor of the conversion endpoint application.
+ * @param srv a service container
+ */
+ConversionApp::ConversionApp (cppcms::service & srv) : cppcms::application (srv)
+{
+	dispatcher ().assign ("/formats", &ConversionApp::formats, this);
+	mapper ().assign ("formats", "/formats");
+
+	dispatcher ().assign ("", &ConversionApp::convert, this);
+	mapper ().assign ("");
+}
+
+/**
+ * @brief handler for the conversion resource.
+ *        takes a snippet and configuration format as input and converts the
+ *		  snippet into a desired output format.
+ */
+void ConversionApp::convert ()
+{
+	RootApp::setCORSHeaders (response (), "POST,OPTIONS");
+
+	if (request ().request_method () == "POST")
+	{
+		// check if request data is of type application/json
```

Comment:
It is okay if you use assert+logs only where they are most urgent (and useful). They can be added at other places later (if needed, usually people tend to overuse logs anyway).